core/src/cost_model.rs

```diff
@@ -90,6 +108,30 @@ impl CostModel {
         cost
     }
 
+    pub fn calculate_cost_no_alloc(&self, transaction: &Transaction, cost: &mut TransactionCost) {
+        cost.reset();
```

Comment:
Can you please add comments that this function is throwing out existing contents of `cost` and the argument is only  used to reuse already allocated memory?

Reply:
added comments